src/components/Calculator.js

```diff
@@ -1,39 +1,84 @@
 import React from 'react';
 import Key from './InputCalc';
+import calculate from './logic/calculate';
 
 class Calculator extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.retrieveSymbol = this.retrieveSymbol.bind(this);
+    this.state = {
+      display: '0',
+      total: null,
+      next: null,
+      operation: null,
+    };
+  }
+
+  async retrieveSymbol(symbol) {
+    const numbers = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9', '.'];
+    const operators = ['%', '÷', 'x', '-', '+'];
+    const {
+      display, total, next, operation,
+    } = this.state;
+
+    console.log(symbol);
+
+    if (numbers.includes(symbol)) {
+      if (display === '0') {
+        this.setState({ display: symbol, total: symbol });
+      } else if (total === null) {
+        this.setState({ display: display + symbol, total: symbol });
+      } else { this.setState({ display: display + symbol, total: total + symbol }); }
+    } else if (operators.includes(symbol)) {
+      this.setState({
+        operation: symbol,
+        next: total,
+        total: null,
+        display: `${display} ${symbol} `,
+      });
+    } else if (symbol === '=') {
+      this.setState(calculate({ total, next, operation }, symbol));
+      this.setState({ display: total });
+    } else {
+      this.setState(calculate({ total, next, operation }, symbol));
+      this.setState({ display: total });
+    }
+    console.log(this.state);
+  }
+
```

Comment:
- [ ] The calculations are not working as expected. Please take a look at the gif below to understand the point am trying to make here.

Gif of Demo Test|
:--------------------:|
![math](https://user-images.githubusercontent.com/39523312/158730921-3b3461f5-75c0-4b76-b677-395b9111f1ca.gif)

I will suggest you work on ensuring that the calculator's result is accurate, and can perform all math operations. 

Also, try making use of lifecycle methods.  I think they might be useful in these operations.